packages/components/src/date-time/time.js

```diff
@@ -120,46 +180,129 @@ class TimePicker extends Component {
 		this.setState( { minutes: event.target.value } );
 	}
 
+	maxHours() {
+		return ( this.props.is12Hour ? 12 : 23 );
+	}
+
+	minHours() {
+		return ( this.props.is12Hour ? 1 : 0 );
+	}
+
 	render() {
 		const { is12Hour } = this.props;
-		const { minutes, hours, am } = this.state;
+		const { day, month, year, minutes, hours, am } = this.state;
 
 		return (
-			<div className="components-time-picker">
-				<input
-					className="components-time-picker__input"
-					type="text"
-					value={ hours }
-					onChange={ this.onChangeHours }
-					onBlur={ this.updateHours }
-				/>
-				<span className="components-time-picker__separator">:</span>
-				<input
-					className="components-time-picker__input"
-					type="text"
-					value={ minutes }
-					onChange={ this.onChangeMinutes }
-					onBlur={ this.updateMinutes }
-				/>
-				{ is12Hour && <div>
-					<Button
-						isDefault
-						className="components-time-picker__am-button"
-						isToggled={ am === 'AM' }
-						onClick={ this.updateAmPm( 'AM' ) }
-					>
-						{ __( 'AM' ) }
-					</Button>
-					<Button
-						isDefault
-						className="components-time-picker__pm-button"
-						isToggled={ am === 'PM' }
-						onClick={ this.updateAmPm( 'PM' ) }
-					>
-						{ __( 'PM' ) }
-					</Button>
-				</div> }
-			</div>
+			<fieldset
+				className={
+					is12Hour ? 'components-datetime__time components-datetime__time--12-hour' : 'components-datetime__time components-datetime__time--24-hour'
+				}
+			>
+				<legend>{ __( 'Publish at' ) }</legend>
+				<fieldset className="invisible">
+					<legend className="invisible">{ __( 'Date' ) }</legend>
+					<div className="components-datetime__time__form">
+						<div className="components-datetime__time__form__container components-datetime__time__form__container--month">
+							<select
+								aria-label={ __( 'Month' ) }
+								name="month"
+								value={ month }
+								onChange={ this.onChangeMonth }
+								onBlur={ this.updateMonth }
+							>
+								<option value="01">{ __( 'January' ) }</option>
+								<option value="02">{ __( 'February' ) }</option>
+								<option value="03">{ __( 'March' ) }</option>
+								<option value="04">{ __( 'April' ) }</option>
+								<option value="05">{ __( 'May' ) }</option>
+								<option value="06">{ __( 'June' ) }</option>
+								<option value="07">{ __( 'July' ) }</option>
+								<option value="08">{ __( 'August' ) }</option>
+								<option value="09">{ __( 'September' ) }</option>
+								<option value="10">{ __( 'October' ) }</option>
+								<option value="11">{ __( 'November' ) }</option>
+								<option value="12">{ __( 'December' ) }</option>
+							</select>
+						</div>
+						<div key="day-field" className="components-datetime__time__form__container components-datetime__time__form__container--day">
```

Comment:
Why do we need a `key` ?

There are a few other instances of this in the component where the same comment applies.

Reply:
Dunno, but I nixed them and it seems fine.